applications/IgaApplication/custom_processes/snake_sbm_process.h

```diff
@@ -0,0 +1,256 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Nicolo' Antonelli
+//                   Andrea Gorgi
+//
+
+#pragma once
+
+// Project includes
+#include "containers/model.h"
+#include "includes/model_part.h"
+#include "spatial_containers/bins_dynamic.h"
+#include "processes/process.h"
+
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/**
+ * @class SnakeSbmProcess
+ * @brief Process class for implementing Snake-based Surrogate Boundary Method (SBM).
+ * This class provides various functions to create and manipulate surrogate boundaries
+ * for Iga models in Kratos.
+ */
+class KRATOS_API(IGA_APPLICATION) SnakeSbmProcess
+    : public Process
+{
+
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of SnakeSbmProcess
+    KRATOS_CLASS_POINTER_DEFINITION(SnakeSbmProcess);
+
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor
+    SnakeSbmProcess(
+        Model& rModel,
+        Parameters ThisParameters);
+
+    /// Destructor.
+    ~SnakeSbmProcess() = default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Execute() override
+    {};
+    
+    void ExecuteInitialize() override
+    {};
+
+    void ExecuteInitializeSolutionStep() override
+    {};
+
+    void ExecuteBeforeSolutionLoop() override
+    {};
+
+    const Parameters GetDefaultParameters() const override
+    {
+        const Parameters default_parameters = Parameters(R"(
+        {
+            "iga_model_part_name" : "IgaModelPart",
```

Comment:
```suggestion
            "model_part_name" : "",
```
Note: 
1. `model_part_name`  to be consistent with the other Kratos processes
2. We provide a default empty value so we can check if the user has provided this field (i.e., we throw an error if after the default validations the `model_part_name` value is an empty string).